kodeagent.py

```
@@ -1686,8 +1687,9 @@ def __init__(
         # TODO Somehow dynamically identify and include the modules used by the tools
         self.tools_source_code: str = 'from typing import *\n\nimport kutils as ku\n\n'
 
-        for t in self.tools:
-            self.tools_source_code += inspect.getsource(t).strip('@tool') + '\n'
+        if tools:
+            for t in self.tools:
+                self.tools_source_code += inspect.getsource(t).strip('@tool') + '\n'
```

Ruff (0.11.9)

Using `.strip()` with multi-character strings is misleading

(B005)